specs/online_order_spec.rb

```diff
@@ -1,90 +1,161 @@
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org", {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip_code=>"98872-9105"})
+
+      prohibited_statuses = [:processing, :shipped, :complete]
+      prohibited_statuses.each do |status|
+        online_order = Grocery::OnlineOrder.new(1325, products, customer, status)
+
+        online_order.add_product("salad", 4.25).must_raise StandardError
+      end
     end
+
   end
 
   describe "OnlineOrder.all" do
     it "Returns an array of all online orders" do
-      # TODO: Your test code here!
+      online_orders_entered = Grocery::OnlineOrder.all
+
+      online_orders_entered.class.must_equal Array
+      count = 0
+      online_orders_entered.each do |single_order|
+        count += 1
+      end
+      count.must_equal 100
     end
 
     it "Returns accurate information about the first online order" do
-      # TODO: Your test code here!
+      online_orders_entered = Grocery::OnlineOrder.all
+      first_order = online_orders_entered[0]
+
+      first_order.id.must_equal 1
+      first_order.products.must_equal({"Lobster"=>17.18, "Annatto seed"=>58.38, "Camomile"=>83.21})
+      first_order.customer.id.must_equal 25
+      first_order.status.must_equal :complete
     end
 
     it "Returns accurate information about the last online order" do
-      # TODO: Your test code here!
+      online_orders_entered = Grocery::OnlineOrder.all
+
+      last_order = online_orders_entered[-1]
+
+      last_order.id.must_equal 100
+      last_order.products.must_equal({"Amaranth"=>83.81, "Smoked Trout"=>70.6, "Cheddar"=>5.63})
+      last_order.customer.id.must_equal 20
+      last_order.status.must_equal :pending
     end
   end
 
   describe "OnlineOrder.find" do
     it "Will find an online order from the CSV" do
-      # TODO: Your test code here!
+      specific_order = Grocery::OnlineOrder.find(5)
+
+      specific_order.id.must_equal 5
+      specific_order.products.must_equal({"Sour Dough Bread"=>35.11})
+      specific_order.customer.id.must_equal 21
+      specific_order.status.must_equal :shipped
     end
 
     it "Raises an error for an online order that doesn't exist" do
-      # TODO: Your test code here!
+      fake_order = Grocery::OnlineOrder.find(500)
+
+      fake_order.must_raise StandardError
     end
   end
 
   describe "OnlineOrder.find_by_customer" do
     it "Returns an array of online orders for a specific customer ID" do
-      # TODO: Your test code here!
+      customers_online_orders = Grocery::OnlineOrder.find_by_customer(15)
+
+      customers_online_orders.length.must_equal 3
+      customers_online_orders.class.must_equal Array
     end
 
     it "Raises an error if the customer does not exist" do
-      # TODO: Your test code here!
+      customers_online_orders = Grocery::OnlineOrder.find_by_customer(150)
```

Comment:
In conjunction with my note about `.find` you should test raising errors like this:
```ruby
proc {
  Grocery::OnlineOrder.find_by_customer(150)
}.must_raise StandardError
```

That said this is advanced and it's good that you were trying it.